src/services/WaveformDataUtils.js

```diff
@@ -259,40 +303,30 @@ export default class WaveformDataUtils {
   /**
    * Prevent the times of segment being edited overlapping with the existing segments
    * @param {Object} segment - segement being edited in the waveform
+   * @param {Boolean} inMarker - true -> start time changed, false -> end time changed
    * @param {Object} peaksInstance - current peaks instance for waveform
    */
-  validateSegment(segment, peaksInstance) {
-    const allSegments = peaksInstance.segments.getSegments();
+  validateSegment(segment, inMarker, peaksInstance) {
     const duration = this.roundOff(peaksInstance.player.getDuration());
 
     const { startTime, endTime } = segment;
 
     // segments before and after the editing segment
-    const { before, after } = this.findWrapperSegments(segment, allSegments);
+    const { before, after } = this.findWrapperSegments(segment, peaksInstance);
 
-    for (let i = 0; i < allSegments.length; i++) {
-      const current = allSegments[i];
-      if (current.id == segment.id) {
-        continue;
+    if (inMarker) {
+      if (before && startTime < before.endTime) {
+        segment.update({ startTime: before.endTime });
+      } else if (startTime > endTime) {
+        segment.update({ startTime: segment.endTime - 0.001 });
       }
-      if (startTime > current.startTime && endTime < current.endTime) {
-        segment.startTime = current.endTime;
-        segment.endTime = current.endTime + 0.001;
-      } else if (
-        duration - 0.001 <= endTime &&
-        endTime <= duration &&
-        after &&
-        after.id === current.id
-      ) {
-        segment.endTime = after.startTime;
-      } else if (startTime > current.startTime && startTime < current.endTime) {
-        segment.startTime = current.endTime;
-      } else if (endTime > current.startTime && endTime < current.endTime) {
-        segment.endTime = current.startTime;
-      } else if (segment.startTime === segment.endTime) {
-        segment.endTime = segment.startTime + 0.001;
+    } else {
+      if (after && endTime > after.startTime) {
+        segment.update({ endTime: after.startTime });
```

Comment:
Am I reading this correctly that if you try to make the end time later then it will force it back to after.startTime?

Reply:
Yes, the same as before this won't go past after.startTime.